axum-core/src/extract/mod.rs

```diff
@@ -204,8 +204,9 @@ impl<B> RequestParts<B> {
     /// Gets a mutable reference to the request body.
     ///
     /// Returns `None` if the body has been taken by another extractor.
-    pub fn body_mut(&mut self) -> Option<&mut B> {
-        self.body.as_mut()
+    // this returns `&mut Option<B>` rather than `Option<&mut B>` such that users can swap it
```

Comment:
Missing period and I think "swap it" is a bit unclear, maybe something like this?

```suggestion
    // this returns `&mut Option<B>` rather than `Option<&mut B>` such that users can use it to set the body.
```